blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticIO.java

```diff
@@ -145,25 +145,29 @@ private IndexRequestBuilder createSingleRequest(Discovery md) throws IOException
 
     public List<SearchResult> search(String tenant, List<String> queries) throws Exception {
         List<SearchResult> results = new ArrayList<SearchResult>();
+
         Timer.Context multiSearchCtx = searchTimer.time();
         queryBatchHistogram.update(queries.size());
-        BoolQueryBuilder qb = boolQuery();
+        BoolQueryBuilder bqb = boolQuery();
+        QueryBuilder qb;
 
         for (String query : queries) {
-            qb.should(boolQuery()
+            GlobPattern pattern = new GlobPattern(query);
+            if (!pattern.hasWildcard()) {
+                qb = termQuery(metric_name.name(), query);
+            } else {
+                qb = regexpQuery(metric_name.name(), pattern.compiled().toString());
```

Comment:
That is a good question and I frankly do not know the answer to that entirely. All I can say is, the current api will become richer, but it will not change the current scheme of things. The part which I am unclear about is whether that will need a endpoint version change...